operator/controllers/kyma_controller.go

```diff
@@ -93,8 +85,12 @@ func (r *KymaReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 	kyma = *kyma.DeepCopy()
 
 	// check if deletionTimestamp is set, retry until it gets fully deleted
-	if !kyma.DeletionTimestamp.IsZero() && kyma.Status.State != operatorv1alpha1.KymaStateDeleting {
-		return ctrl.Result{}, r.updateKymaStatus(ctx, &kyma, operatorv1alpha1.KymaStateDeleting, "deletion timestamp set")
+	if !kyma.DeletionTimestamp.IsZero() {
+		// if the status is not yet set to deleting, also update the status
+		if kyma.Status.State != operatorv1alpha1.KymaStateDeleting {
+			return ctrl.Result{}, r.updateKymaStatus(ctx, &kyma, operatorv1alpha1.KymaStateDeleting, "deletion timestamp set")
+		}
```

Comment:
this is not required, reconciliation wont be triggered if update status is redundant